### PR TITLE
feat(titanium): update to latest SDK 12.0.0

### PR DESCRIPTION
**Pull request by m1ga**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/tidev/titanium_mobile/tree/master/apidoc
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by cb1kenobi — APPROVED** [review]

File: types/titanium/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "rules": {
+        "no-any-union": "off",
+        "no-unnecessary-generics": "off",
+        "prefer-declare-function": "off"
```
Comment (jakebailey) [inline types/titanium/.eslintrc.json:5]: Can you provide some examples/rationale as to why these need to be disabled? Each of these should be flagging real issues.

**Review by jakebailey — CHANGES_REQUESTED** [review]

File: types/titanium/.eslintrc.json (hunk 1 above)
Comment (m1ga) [reply types/titanium/.eslintrc.json:5]: the test showed an error for every file:
```
/DefinitelyTyped/types/titanium/base.d.ts
  1:1  error  Definition for rule 'no-any-union' was not found             no-any-union
  1:1  error  Definition for rule 'no-unnecessary-generics' was not found  no-unnecessary-generics
  1:1  error  Definition for rule 'prefer-declare-function' was not found  prefer-declare-function
```
so I've created that file

Comment (jakebailey) [reply types/titanium/.eslintrc.json:5]: In CI? Or locally? It seems to me like you just needed to update main and update your copy of dtslint.

Comment (m1ga) [reply types/titanium/.eslintrc.json]: locally, when I run the last step in the todo list.
But you are right! Updating master and running the test again without the file worked fine too :+1:  Thanks for the hint

**Comment by jakebailey** [comment]
Thanks for the removal; this still needs owner approval if possible.

**Review by janvennemann — APPROVED** [review]

**Review by cb1kenobi — APPROVED** [review]

**Review by jakebailey — APPROVED** [review]

**Comment by m1ga** [comment]
Ready to merge

**Pull request merged**